code/data_setup/parse_alov_bb.py

```diff
@@ -0,0 +1,68 @@
+"""A script for parsing the Alov bounding box `.ann` files."""
+
+import sys
+import os
+import itertools
+import pandas as pd
+
+def parse_file(bbox_dir, filename): 
+    """Parse an individual `.ann` file and output the relevant elements.
+
+    Args: 
+    ----
+        bbox_dir: str
+            Holds the directory path to the `.ann` file. 
+        filename: str
+            Holds the name of the `.ann` file. 
+
+    Output: 
+    ------
+        output_lst: list of lists 
+            Holds the elements grabbed from the `.ann` file. 
+    """
+
+    output_lst = []
+
+    with open(bbox_dir + filename) as f: 
+        output_lst = [parse_line(filename, line) for line in f]
+
+    return output_lst 
+
+def parse_line(filename, line): 
+    """Parse an individual line from an `.ann` file. 
+
+    Args: 
+    ----
+        filename: str
+        line: str
+
+    Output: 
+    ------
+        object_lst: list
+            Parsed list of elements. 
+    """
+
+    filename_parts = filename.split('_')
+    line_parts = line.split()
+    object_lst = [filename_parts[0], filename, line_parts[0], 
+            line_parts[1], line_parts[2], line_parts[3], 
+            line_parts[4], line_parts[5], line_parts[6], 
+            line_parts[7], line_parts[8]]
+
```

Comment:
So, I thought about this route, but then out of every function your outputting a `DataFrame`, or `numpy array` (if you call `.values`). In the first case, I imagine you could write to your output file in an append mode, and in the second you'd have to concat some numpy arrays ( I guess you could also concat dataframe objects if you wanted). Are either of these definitively better options (they might be, I just didn't see them as clearly better)?